### ISSUE TITLE
Error on https://game1164.konggames.com/games/MateuszSkutnik/submachine-2-the-lighthouse/frame/61ddf6aa-48e9-4379-8b40-b9f12c1e6b58/

### ISSUE THREAD

**Issue by jms8142** [ruffle-rs/ruffle#11823]
# Error Info
Error name: Error
Error message: panicked at 'index out of bounds: the len is 2 but the index is 2', /home/runner/.cargo/git/checkouts/nihav-vp6-660fee9efdd4622a/9416fcc/nihav-duck/src/codecs/vp56.rs:1032:12
Error stack:
```
Error: panicked at 'index out of bounds: the len is 2 but the index is 2', /home/runner/.cargo/git/checkouts/nihav-vp6-660fee9efdd4622a/9416fcc/nihav-duck/src/codecs/vp56.rs:1032:12
    at W.n.wbg.__wbg_new_d258248ed531ff54 (https://unpkg.com/@ruffle-rs/ruffle/core.ruffle.079106c39c4e7b9aa976.js:1:51548)
    at core::ops::function::FnOnce::call_once{{vtable.shim}}::h02df5e9c2be9a573 (wasm://wasm/02ce7e4a:wasm-function[8732]:0x858ef5)
    at std::panicking::rust_panic_with_hook::h85b7f6628c291e12 (wasm://wasm/02ce7e4a:wasm-function[5685]:0x77e246)
    at core::panicking::panic_fmt::hf5c4cd929d4aaa9e (wasm://wasm/02ce7e4a:wasm-function[8594]:0x854a70)
    at core::panicking::panic_bounds_check::h0cfce459260a0a82 (wasm://wasm/02ce7e4a:wasm-function[6862]:0x7d8aec)
    at nihav_duck::codecs::vp56::VP56Decoder::predict_dc::h2d58dcdb9ffd938a (wasm://wasm/02ce7e4a:wasm-function[1901]:0x4dc964)
    at nihav_duck::codecs::vp56::VP56Decoder::decode_planes::hfc504ef38713c8a0 (wasm://wasm/02ce7e4a:wasm-function[527]:0xa8d04)
    at <ruffle_video_software::decoder::vp6::Vp6Decoder as ruffle_video_software::decoder::VideoDecoder>::decode_frame::h08672e98999b8b7a (wasm://wasm/02ce7e4a:wasm-function[675]:0x214ec4)
    at <ruffle_video_software::backend::SoftwareVideoBackend as ruffle_video::backend::VideoBackend>::decode_video_stream_frame::hbe189597894e70c9 (wasm://wasm/02ce7e4a:wasm-function[2356]:0x56d4db)
    at ruffle_core::display_object::video::Video::seek::hce8d00237384ccd2 (wasm://wasm/02ce7e4a:wasm-function[712]:0x246242)
```

# Player Info
Allows script access: true
Renderer: wgpu
Adapter Backend: Gl
Adapter Name: "ANGLE (Intel Inc., Intel(R) Iris(TM) Plus Graphics 645, OpenGL 4.1)"
Adapter Device Type: IntegratedGpu
Adapter Driver Name: ""
Adapter Driver Info: ""
Enabled features: Features(TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES | PUSH_CONSTANTS)
Available features: Features(CLEAR_TEXTURE)
Current limits: Limits { max_texture_dimension_1d: 16384, max_texture_dimension_2d: 16384, max_texture_dimension_3d: 2048, max_texture_array_layers: 256, max_bind_groups: 4, max_bindings_per_bind_group: 640, max_dynamic_uniform_buffers_per_pipeline_layout: 8, max_dynamic_storage_buffers_per_pipeline_layout: 0, max_sampled_textures_per_shader_stage: 16, max_samplers_per_shader_stage: 16, max_storage_buffers_per_shader_stage: 0, max_storage_textures_per_shader_stage: 0, max_uniform_buffers_per_shader_stage: 11, max_uniform_buffer_binding_size: 16384, max_storage_buffer_binding_size: 0, max_vertex_buffers: 8, max_buffer_size: 268435456, max_vertex_attributes: 16, max_vertex_buffer_array_stride: 255, min_uniform_buffer_offset_alignment: 256, min_storage_buffer_offset_alignment: 256, max_inter_stage_shader_components: 60, max_compute_workgroup_storage_size: 0, max_compute_invocations_per_workgroup: 0, max_compute_workgroup_size_x: 0, max_compute_workgroup_size_y: 0, max_compute_workgroup_size_z: 0, max_compute_workgroups_per_dimension: 0, max_push_constant_size: 96 }
Surface quality: high
Surface samples: 4
Surface size: Extent3d { width: 550, height: 400, depth_or_array_layers: 1 }

# Page Info
Page URL: https://game1164.konggames.com/games/MateuszSkutnik/submachine-2-the-lighthouse/frame/61ddf6aa-48e9-4379-8b40-b9f12c1e6b58/?kongregate_host=www.kongregate.com
SWF URL: https://game1164.konggames.com/gamez/0000/1164/live/submachine_2.swf?kongregate_game_version=1396219602

# Browser Info
User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36
Platform: MacIntel
Has touch support: false

# Ruffle Info
Version: 0.1.0
Name: nightly 2023-07-01
Channel: nightly
Built: 2023-07-01T00:25:59.437Z
Commit: 29a5c6a4cd9918d5e1eb89cc42ccd03d4e920931
Is extension: false

# Metadata
width: 400
height: 420
frameRate: 25
numFrames: 31
swfVersion: 10
backgroundColor: #000000
isActionScript3: false
uncompressedLength: 5918788

**Comment by torokati44** [comment]
Same as #11822, I bisected this regression to c86360a1, merged as part of #10756.

**Comment by Lord-McSweeney** [comment]
Fixed by #11831.
